packages/cache-inmemory/src/services/inmemory-cache.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  ConfigurableModuleDeclaration,
+  MODULE_RESOURCE_TYPE,
+  TransactionBaseService,
+  ICacheService,
+} from "@medusajs/medusa"
+import { MedusaError } from "medusa-core-utils"
+import { EntityManager } from "typeorm"
+
+import { CacheRecord, InMemoryCacheModuleOptions } from "../types"
+
+const DEFAULT_TTL = 30 // seconds
+
+type InjectedDependencies = {}
+
+/**
+ * Class represents basic, in-memory, cache store.
+ */
+class InMemoryCacheService
+  extends TransactionBaseService
```

Comment:
question: Not sure it is necessary to extends this class and hold the manager in the class. 
If some provide a db cache, then he will have a local transaction and if the core fail the result will still be cached which I think still makes sense if it succeed to compute something. But overall I am not sure we need that